### ISSUE TITLE
Code Table Request - Albuquerque Biopark Zoo Local ID

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#5772]
## Instructions

This is a template to facilitate communication with the Arctos Code Table Committee. Submit a separate request for each relevant value. This form is appropriate for exploring how data may best be stored, for adding vocabulary, or for updating existing definitions.

Reviewing documentation before proceeding will result in a more enjoyable experience.

* [Issue Documentation](http://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html)
* [Code Table Documentation](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html)
* [Video Tutorial - Submit a Code Table Request](https://youtu.be/t2jHbsRA3lk)

------------------------------

## Initial Request

**Goal**: Describe what you're trying to accomplish. This is the only necessary step to start this process. The Committee is available to assist with all other steps. Please clearly indicate any uncertainty or desired guidance if you proceed beyond this step.

Replace ambiguous other ID with generic other ID type and issued by agent

> All  [[Albuquerque Biopark Zoo Local ID ](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#albuquerque_biopark_zoo_local_id) should be replaced with other ID type = other identifier and issued by agent [ABQ BioPark](https://arctos.database.museum/agents.cfm?agent_id=1008931)

**Proposed Value**: Proposed new value. This should be clear and compatible with similar values in the relevant table and across Arctos. 

N/A

**Proposed Definition**: Clear, complete, non-collection-type-specific **functional** definition of the value. Avoid discipline-specific terminology if possible, include parenthetically if unavoidable.

N/A

**Context**: Describe why this new value is necessary and existing values are not.
See #4604 

**Table**: Code Tables are http://arctos.database.museum/info/ctDocumentation.cfm. Link to the specific table or value. This may involve multiple tables and will control datatype for Attributes. OtherID requests require BaseURL (and example) or explanation. Please ask for assistance if unsure.

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type

**Collection type**: Some code tables contain collection-type-specific values. ``collection_cde`` may be found from https://arctos.database.museum/home.cfm

N/A

**Priority**: Please describe the urgency and/or choose a priority-label to the right. You should expect a response within two working days, and may utilize [Arctos Contacts](https://arctosdb.org/contacts/) if you feel response is lacking.

Normal

**Available for Public View**: Most data are by default publicly available. Describe any necessary access restrictions. 

**Project**: Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

**Discussion**: Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

## Approval

All of the following must be checked before this may proceed.

The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality).

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [x] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection

If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc. 

1. Can a suitable solution be found here? If not, proceed to (2)
2. Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)
3. Take the discussion to a monthly Arctos Working Group meeting for final resolution.

## Implementation

Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made.

- [ ] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel. 

- [ ] Make changes as described above. Ensure the URL of this Issue is included in the definition. 

Close this Issue.

**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result.

## Special Exemptions

In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases.

1. Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step.
2. The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology.

**Comment by dustymc** [comment]
I will plan on proceeding with this about 2023-03-23 if there are no objections.

I will proceed immediately upon approval of each of the involved collections.


Data:

[temp_albuquerque_biopark_zoo_local_id.csv.zip](https://github.com/ArctosDB/arctos/files/10934434/temp_albuquerque_biopark_zoo_local_id.csv.zip)

Summary:

```
  guid_prefix  | numrecs 
---------------+---------
 Arctos:Entity |       1
 MSB:Bird      |       1
 MSB:Herp      |      10
 MSB:Mamm      |    4659

```


Users:


 @campmlc
 @jtgiermakowski
 @AdrienneRaniszewski
 @jldunnum
 @catherpes
 @Jegelewicz


See also https://github.com/ArctosDB/arctos/issues/5771

**Comment by campmlc** [comment]
Not OK to proceed. https://github.com/ArctosDB/arctos/issues/6004 https://github.com/ArctosDB/arctos/issues/6002

**Comment by Jegelewicz** [comment]
@campmlc what do we need to do to make this work?

**Comment by campmlc** [comment]
This needs to be left in the code table for now. Too much important data hanging on it requiring dedicated attention that can't be provided at the moment.

**Comment by mkoo** [comment]
Not sure I understand the nature of "Too much important data hanging on it" as currently you are in halfway there.
See entry and identifiers associated with https://arctos.database.museum/agent/1008931

I think Teresa and Dusty can help with this since you dont have time. Polish the agent page, add any appropriate AKA, move to Issued By and you will have a better way to find all the ABQ BioPark ID. If needed, there could be an additional "ABQ BioPark Local ID" agent entry (and not use the one linked above) where you can eventually drop in the relevant details (e.g.,  if it's a separate catalog etc). Right now, you have no place to put any metadata if it stays in the Code Table.

**Comment by campmlc** [comment]
This identifier cannot be merged at this time.  It is not just me saying this - @AdrienneRaniszewski @jldunnum .
The Biopark agent issues more than one identifier. We use the Local ID vs the GAN number to make entities and track individual animals sampled over time. ANY agent that issues more than one identiifer should not have all their identifier types merged into other identifier. Please, can we postpone these until we can meet and discuss in person, when we have dedicated time to discuss? The timing couldn't be worse right now.

**Comment by campmlc** [comment]
Again, to reiterate - MSB uses this identifier in the same way we use NK. We don't want the identifier converted to other identifier issued by Albuquerque Biopark. We do not want to create an agent of Albuquerque Biopark Local ID. So please leave for now.

**Comment by Jegelewicz** [comment]
@campmlc I am looking for some clarity because I don't understand the opposition to the new format for identifiers.

I loaded all the ABQ Biopark identifiers using the new format, but I left the current ones and I can remove the new if necessary. I'd like your help testing the new format so that I can understand what it is not doing that the old format did. If we can understand what is missing, we can make adjustments to the model if needed!

Also note that the GAN is not issued by ABQ Biopark, so there shouldn't be any confusion between those two numbers. The GAN is generated by ZIMS, which has to coordinate globally to ensure there aren't duplicate organism identifiers.

I know that the agent Albuquerque Biopark Local ID feels odd, but it is so much more powerful that the simple description allowed by the code table! All I am asking is that you give this a chance and provide me with feedback on what doesn't work with this model. I'd really appreciate it!

**Comment by Jegelewicz** [comment]
> I loaded all the ABQ Biopark identifiers using the new format, but I left the current ones and I can remove the new if necessary. I'd like your help testing the new format so that I can understand what it is not doing that the old format did. If we can understand what is missing, we can make adjustments to the model if needed!
> 
> All I am asking is that you give this a chance and provide me with feedback on what doesn't work with this model. I'd really appreciate it!

The identifiers I added were removed and this issued closed without any testing. I have reopened the issue and loaded the identifiers in test as requested. @dustymc I don't think the component loaders are running in test - can you make these load? https://arctos-test.tacc.utexas.edu/loaders/BulkloadOtherId.cfm#jegelewicz

**Comment by campmlc** [comment]
I'd like to repeat my request to change all these to "Local ID" as this ID is used by many different zoos and endangered species recovery programs, and we could use this simpler more generic identifier with an "issued by" of the respective agent.  This is far preferable to creating a different agent for each "XXX Zoo Local ID" and "USFWS Captive Breeding Program XXXX Local ID". This identifier type is also a formal ID type in ZIMS.

**Comment by AdrienneRaniszewski** [comment]
Totally agree. It’s a type of identifier. Not an agent. Creating a bunch of new agents to act as identifier types is strange and unintuitive to me. In my view agents are doers. They do things like collect, enter data, or assign identifiers.


~ Adrienne


Collections Associate/Staff Coordinator
MSB Division of Mammals
505-277-5369


From: Mariel Campbell ***@***.***>
Date: Thursday, July 27, 2023 at 1:35 PM
To: ArctosDB/arctos ***@***.***>
Cc: Adrienne Raniszewski ***@***.***>, Mention ***@***.***>
Subject: Re: [ArctosDB/arctos] Code Table Request - Albuquerque Biopark Zoo Local ID (Issue #5772)
  [EXTERNAL]

I'd like to repeat my request to change all these to "Local ID" as this ID is used by many different zoos and endangered species recovery programs, and we could use this simpler more generic identifier with an "issued by" of the respective agent. This is far preferable to creating a different agent for each "XXX Zoo Local ID" and "USFWS Captive Breeding Program XXXX Local ID". This identifier type is also a formal ID type in ZIMS.

**Comment by mkoo** [comment]
This https://github.com/ArctosDB/arctos/issues/6453 could be essentially a compromise to this original request.
 I am adding this here so we can track the two but this has become a local collection issue which was discussed directly

**Comment by campmlc** [comment]
> This #6453 could be essentially a compromise to this original request. I am adding this here so we can track the two but this has become a local collection issue which was discussed directly

I support this the solution proposed in #6453.
